tests/test_dates.py

```diff
@@ -24,6 +24,23 @@
 from babel.util import FixedOffsetTimezone
 
 
+@pytest.fixture(params=["pytz.timezone", "zoneinfo.ZoneInfo"])
+def timezone_getter(request):
+    if request.param == "pytz.timezone":
+        return timezone
+    elif request.param == "zoneinfo.ZoneInfo":
+        try:
+            import zoneinfo
+        except ImportError:
+            try:
+                from backports import zoneinfo
+            except ImportError:
+                pytest.skip("zoneinfo not available")
+        return zoneinfo.ZoneInfo
+    else:
+        raise NotImplementedError
```

Comment:
How about something like this, to precompute the tz implementations we have? This way tests wouldn't end up skipped half of the time.

```suggestion

tz_implementations = {"pytz.timezone": timezone}
try:
    import zoneinfo
except ImportError:
    try:
        from backports import zoneinfo
    except ImportError:
        zoneinfo = None
if zoneinfo:
     tz_implementations["zoneinfo.ZoneInfo"] = zoneinfo
      
@pytest.fixture(params=list(tz_implementations))
def timezone_getter(request): 
    return tz_implementations[request.param]
```

Reply:
@akx, can you check my answer and tell me if you then prefer to have them showing as skipped or not?

Reply:
Sorry for the delay. Yeah, you do have a point with the skipped tests there – we can probably leave this as-is.